server/src/main/java/org/opensearch/action/support/replication/TransportWriteAction.java

```diff
@@ -514,15 +515,20 @@ public void failShardIfNeeded(
             if (TransportActions.isShardNotAvailableException(exception) == false) {
                 logger.warn(new ParameterizedMessage("[{}] {}", replica.shardId(), message), exception);
             }
-            shardStateAction.remoteShardFailed(
-                replica.shardId(),
-                replica.allocationId().getId(),
-                primaryTerm,
-                true,
-                message,
-                exception,
-                listener
-            );
+            // If a write action fails due to the closure of the primary shard
+            // then the replicas should not be marked as failed since they are
+            // still up-to-date with the (now closed) primary shard
+            if (exception instanceof PrimaryShardClosedException == false) {
```

Comment:
For my understanding, I am guessing we would not successfully ack back request to the client and eventually the replica copy would get promoted to primary and the request subsequently retried on the newly promoted copy.
Can we add an integration test in ClusterDisruptionIT to verify this behaviour?